### PR TITLE
feat(sil_euro_latin): add Caps Lock and Start-of-Sentence detection

### PR DESCRIPTION
**Pull request by mcdurdin**
Caps Lock and Start-of-Sentence detection are new features for touch devices in Keyman 15.0. The compiled keyboard remains backwardly compatible with older versions of Keyman, as access to the new functionality will just be ignored (this is known as progressive enhancement). However, the keyboard must be built with the 15.0 compiler.

In order to test the new functionality, ensure you are running Keyman 15.0.265 or later.

### CODE REVIEW

**Comment by mcdurdin** [comment]
> Just in case, I would like to see a user test against Keyman 14 as well as 15.

* [sil_euro_latin.zip](https://github.com/keymanapp/keyboards/files/9050066/sil_euro_latin.zip)

The .kmp file within the above zip contains the changes in this PR, built with kmcomp 15.0.265. I am testing against Keyman 14 and 15 for Android currently, and will test on iOS shortly.

**Comment by mcdurdin** [comment]
I have tested on:

* Keyman for Windows 14.0.294 - no issues encountered
* Keyman for Windows 15.0.265 - no issues encountered
* Keyman for Android 14.0.294 - no issues encountered, no change in behaviour from previous version
* Keyman for Android 15.0.265 - Caps Lock and Start-of-sentence on touch working well

File: release/sil/sil_euro_latin/source/sil_euro_latin.kmn [hunk 1]
```diff
@@ -262,6 +270,65 @@ if(option_key = '') + any(controls) > index(controls, 2) dk(1)
...
+
+    c If we are at the start of a text field, we're at the 'start of a
+    c sentence', so we'll move to the Shift layer
+    nul > layer('shift')
+
+    c Otherwise, we'll move to the Shift layer only if we see end of a
+    c sentence followed by one, two, or three spaces.
+    any(sentencePunctuation) any(whitespace) > layer('shift')
+    any(sentencePunctuation) any(whitespace) any(whitespace) > layer('shift')
+    any(sentencePunctuation) any(whitespace) any(whitespace) any(whitespace) > layer('shift')
```
Comment (jahorton) [inline release/sil/sil_euro_latin/source/sil_euro_latin.kmn:326]: Turns out that we'll need a few extra rules here due.  Without loss of generality...

```
any(sentencePunctuation) dk(1) any(whitespace) > layer('shift')
c and extra whitespace variants.
```

Why?

https://github.com/keymanapp/keyboards/blob/919dfff5b8a938bf1784bf6dce19b9a51b8dcac0/release/sil/sil_euro_latin/source/sil_euro_latin.kmn#L34

https://github.com/keymanapp/keyboards/blob/919dfff5b8a938bf1784bf6dce19b9a51b8dcac0/release/sil/sil_euro_latin/source/sil_euro_latin.kmn#L269

When an external keyboard is used on a touch device, there will be an interfering deadkey after the `'.'`.

Unless, _of course_, if the user typed the period key twice in a row:

https://github.com/keymanapp/keyboards/blob/919dfff5b8a938bf1784bf6dce19b9a51b8dcac0/release/sil/sil_euro_latin/source/sil_euro_latin.kmn#L261

**Review by jahorton — CHANGES_REQUESTED** [review]

File: release/sil/sil_euro_latin/source/sil_euro_latin.kmn (hunk 1 above)
Comment (jahorton) [reply release/sil/sil_euro_latin/source/sil_euro_latin.kmn:326]: This was bugging me when testing https://github.com/keymanapp/keyman/pull/6901, as I thought I had a weird bug remaining on my end.  Took a deep drill down to note that there was a deadkey in the way... which led me to the analysis above.

File: release/sil/sil_euro_latin/source/sil_euro_latin.kmn [hunk 2]
```diff
@@ -11,8 +11,8 @@ store(&COPYRIGHT) 'Copyright (c) SIL International'
 $keymanonly: store(&mnemoniclayout) '1'
 
 c KeymanWeb-specific header statements
-store(&KMW_HELPFILE) 'sil_euro_latin.html'
-store(&KMW_EMBEDJS) 'sil_euro_latin_js.txt'
+c store(&KMW_HELPFILE) 'sil_euro_latin.html'
+c store(&KMW_EMBEDJS) 'sil_euro_latin_js.txt'
```
Comment (jahorton) [inline release/sil/sil_euro_latin/source/sil_euro_latin.kmn:15]: Is this part intentional?  Wouldn't this eliminate the custom 'help'-style desktop OSK?

Comment (mcdurdin) [reply release/sil/sil_euro_latin/source/sil_euro_latin.kmn:15]: oops.

**Pull request merged**